pkg/hive/manager.go

```diff
@@ -112,17 +120,17 @@ func (hr *clusterManager) CreateOrUpdate(ctx context.Context, sub *api.Subscript
 	return nil
 }
 
-func (hr *clusterManager) Delete(ctx context.Context, namespace string) error {
-	err := hr.kubernetescli.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{})
+func (hr *clusterManager) Delete(ctx context.Context, doc *api.OpenShiftClusterDocument) error {
+	err := hr.kubernetescli.CoreV1().Namespaces().Delete(ctx, doc.OpenShiftCluster.Properties.HiveProfile.Namespace, metav1.DeleteOptions{})
 	if err != nil && kerrors.IsNotFound(err) {
 		return nil
 	}
 
 	return err
 }
 
-func (hr *clusterManager) IsClusterDeploymentReady(ctx context.Context, namespace string) (bool, error) {
-	cd, err := hr.hiveClientset.HiveV1().ClusterDeployments(namespace).Get(ctx, ClusterDeploymentName, metav1.GetOptions{})
+func (hr *clusterManager) IsClusterDeploymentReady(ctx context.Context, doc *api.OpenShiftClusterDocument) (bool, error) {
+	cd, err := hr.hiveClientset.HiveV1().ClusterDeployments(doc.OpenShiftCluster.Properties.HiveProfile.Namespace).Get(ctx, ClusterDeploymentName, metav1.GetOptions{})
 	if err == nil {
 		for _, cond := range cd.Status.Conditions {
 			if cond.Type == hivev1.ClusterReadyCondition && cond.Status == corev1.ConditionTrue {
```

Comment:
We may want to consider changing this to instead validate that `cd.spec.installed=true` [this](https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/pkg/controller/clusterdeployment/cluster_deployment_controller.go#L199-210) is what we do today for OSD/ROSA.

Reply:
fixed for the Hive install flow